openpathsampling/analysis/move_strategy.py

```diff
@@ -361,182 +398,235 @@ class DefaultStrategy(MoveStrategy):
-            movers=scheme.movers[group],
-            weights=weights
+            movers=mover_weights.keys(),
+            weights=mover_weights.values()
         )
         chooser.name = choosername
         return chooser
 
-
-    def strategy_group_weights(self, scheme, mover_weights=None):
+    def default_weights(self, scheme):
         """
-        Returns the group weights given by info in the strategy and scheme.
+        Set the default weights given the initial `scheme`.
 
-        `mover_weights` is None if either (a) scheme.choice_probability is
-        not set; or (b) self.group_weights is set. Then we don't make use of
-        scheme.choice_probability. If given mover_weights, we use
-        scheme.choice_probability to set the group_weights.
+        Note that this includes preservation of scheme.choice_probability,
+        if it is set.
 
-        See also
-        --------
-        DefaultStrategy.get_weights
-        DefaultStrategy.strategy_mover_weights
+        Parameters
+        ----------
+        scheme : paths.MoveScheme
+            the scheme to which this strategy is being applied
+
+        Returns
+        -------
+        tuple (sortkey_w, movers_w)
+            sortkey_w is a dictionary of sort keys to weights; movers_w is a
+            dictionary of mover keys to weights. See class definition for
+            the specific formats of the keys.
         """
-        group_weights = {}
-        if mover_weights is None:
-            for group in scheme.movers:
+        sortkey_w = {}
+        movers_w = {}
+        if scheme.choice_probability != {}:
+            # extract weights from the choice probability
+            (sortkey_w, movers_w) = self.weights_from_choice_probability(
+                scheme, scheme.choice_probability
+            )
+        else:
+            # generate absolutely generic weights
+            for skey in scheme.movers.keys():
                 try:
-                    group_weights[group] = self.default_group_weights[group]
+                    sortkey_w[skey] = self.default_group_weights[skey]
                 except KeyError:
-                    group_weights[group] = 1.0
-                # override default
-                if group in self.group_weights:
-                    group_weights[group] = self.group_weights[group]
-        else:
-            mover_weights = self.strategy_mover_weights(scheme)
-            group_weights = {}
-            for group in scheme.movers:
-                movers = scheme.movers[group]
-                group_weights[group] = sum([scheme.choice_probability[m] 
-                                            for m in movers])
-            try:
-                normalizer = group_weights['shooting']
-            except KeyError:
-                normalizer = sum(group_weights.values())
-            for group in group_weights:
-                group_weights[group] /= normalizer
+                    sortkey_w[skey] = 1.0
```

Comment:
I found a nice trick to write this in one line.
```
sortkey_w[skey] = self.default_goup_weights.get(key, 1.0)
```
use the `.get(key, default)` method